broker/lib/jobs/BnRStatusPollerJob.js

```diff
@@ -122,11 +132,116 @@ class BnRStatusPollerJob extends BaseJob {
         }
       })
       .then(operationStatusResponse => operationStatusResponse.operationFinished ?
-        this.doPostFinishOperation(operationStatusResponse, operationName, instanceInfo) :
-        Promise.resolve(operationStatusResponse)
+        this.doPostFinishOperation(operationStatusResponse, operationName, instanceInfo) : operationStatusResponse
       );
   }
+
+  static checkOperationCompletionStatus20(job_data) {
+    logger.info('Checking Operation Completion Status for :', job_data);
+    const operationName = job_data.operation;
+    const instanceInfo = job_data.operation_details;
+    const instance_guid = instanceInfo.instance_guid;
+    const backup_guid = instanceInfo.backup_guid;
+    const deployment = instanceInfo.deployment;
+    const plan = catalog.getPlan(instanceInfo.plan_id);
+    return Promise.try(() => {
+        if (operationName === CONST.OPERATION_TYPE.BACKUP) {
+          return lockManager.lock(instance_guid, {
+              lockedResourceDetails: {
+                resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                resourceId: backup_guid,
+                operation: CONST.OPERATION_TYPE.BACKUP
+              }
+            })
+            .catch(err => {
+              if (err instanceof DeploymentAlreadyLocked) {
+                logger.info(`Proceeding as lock is already acquired for the resource: ${instance_guid}`);
+              } else {
+                throw err;
+              }
+            })
+            .then(() => {
+              return BackupManager.createManager(plan)
+                .then(backupManager => backupManager.getOperationState(CONST.OPERATION_TYPE.BACKUP, instanceInfo));
```

Comment:
can actually directly call, agent.getOperatoinState() and avoid depedency with BackupManager.